File: backend/DTOs/TaskDTO/TaskReadDTO.cs
namespace DTOs
{
    public class TaskReadDTO
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public DateTime? Deadline { get; set; }
        public DateTime Created_At { get; set; }
        public string TaskStateName { get; set; } = null!;
        public List<string> TagNames { get; set; } = new();
    }
}

File: frontend/src/app/dtos/task/task-read.dto.ts
import { Type } from 'class-transformer';

export default class TaskReadDTO {
    public readonly id!: string;

    public readonly title!: string;

    public readonly description!: string;

    @Type(() => Date)
    public readonly createdAt!: Date;

    @Type(() => Date)
    public readonly deadline!: Date | null;

    public readonly taskStateName!: string;

    public readonly tagNames!: string[];
}
